File: backend/app/core/strategy_presets.py
from typing import Any, Dict, List

STRATEGY_PRESETS: List[Dict[str, Any]] = [
    {
        "id": "rsi_oversold",
        "name": "RSI Oversold",
        "description": "Buy when RSI is below 30 (Oversold condition)",
        "config": {
            "side": "buy",
            "interval": "1h",
            "conditions": {
                "operator": "<",
                "indicator": "RSI",
                "params": {"period": 14},
                "compare": {"value": 30}
            }
        }
    },
    {
        "id": "macd_crossover",
        "name": "MACD Crossover",
        "description": "Buy when MACD line crosses above the signal line",
        "config": {
            "side": "buy",
            "interval": "1h",
            "conditions": {
                "operator": "crossover",
                "indicator": "MACD",
                "params": {"fast": 12, "slow": 26, "signal": 9},
                "field": "MACD_12_26_9",
                "compare": {
                    "indicator": "MACD",
                    "params": {"fast": 12, "slow": 26, "signal": 9},
                    "field": "MACDs_12_26_9"
                }
            }
        }
    },
    {
        "id": "rsi_macd_combo",
        "name": "RSI + MACD Fusion",
        "description": "Aggressive: RSI < 40 and MACD Crossover",
        "config": {
            "side": "buy",
            "interval": "1h",
            "conditions": {
                "operator": "AND",
                "expressions": [
                    {
                        "operator": "<",
                        "indicator": "RSI",
                        "params": {"period": 14},
                        "compare": {"value": 40}
                    },
                    {
                        "operator": "crossover",
                        "indicator": "MACD",
                        "params": {"fast": 12, "slow": 26, "signal": 9},
                        "field": "MACD_12_26_9",
                        "compare": {
                            "indicator": "MACD",
                            "params": {"fast": 12, "slow": 26, "signal": 9},
                            "field": "MACDs_12_26_9"
                        }
                    }
                ]
            }
        }
    },
    {
        "id": "golden_cross",
        "name": "Golden Cross",
        "description": "Long-term bullish: 50 SMA crosses above 200 SMA",
        "config": {
            "side": "buy",
            "interval": "1d",
            "conditions": {
                "operator": "crossover",
                "indicator": "SMA",
                "params": {"period": 50},
                "compare": {
                    "indicator": "SMA",
                    "params": {"period": 200}
                }
            }
        }
    },
    {
        "id": "bb_reversion",
        "name": "Bollinger Reversion",
        "description": "Buy when price touches the lower Bollinger Band",
        "config": {
            "side": "buy",
            "interval": "15m",
            "conditions": {
                "operator": "<=",
                "field": "close",
                "compare": {
                    "indicator": "BBANDS",
                    "params": {"period": 20, "std": 2.0},
                    "field": "BBL_20_2.0"
                }
            }
        }
    }
]

File: frontend/src/components/strategy-presets.ts
import type { AdvancedCondition } from "./strategy-builder"

export interface StrategyPreset {
  id: string
  name: string
  description: string
  side: "buy" | "sell"
  logicOperator: "AND" | "OR"
  conditions: AdvancedCondition[]
}

export const STRATEGY_PRESETS: StrategyPreset[] = [
  {
    id: "rsi_oversold",
    name: "RSI Oversold",
    description: "Traditional dip-buying: enters when RSI drops below 30.",
    side: "buy",
    logicOperator: "AND",
    conditions: [
      {
        id: "rsi-1",
        indicator: "RSI",
        params: { period: 14 },
        operator: "<",
        targetType: "value",
        targetValue: 30
      }
    ]
  },
  {
    id: "macd_crossover",
    name: "MACD Crossover",
    description: "Momentum shift: MACD line crosses above Signal line.",
    side: "buy",
    logicOperator: "AND",
    conditions: [
      {
        id: "macd-1",
        indicator: "MACD",
        params: { fast: 12, slow: 26, signal: 9 },
        field: "MACD",
        operator: "crossover",
        targetType: "indicator",
        targetIndicator: "MACD",
        targetParams: { fast: 12, slow: 26, signal: 9 },
        targetField: "MACDs"
      }
    ]
  },
  {
    id: "golden_cross",
    name: "Golden Cross",
    description: "Bullish trend: 50-day SMA crosses above 200-day SMA.",
    side: "buy",
    logicOperator: "AND",
    conditions: [
      {
        id: "sma-1",
        indicator: "SMA",
        params: { period: 50 },
        field: "SMA",
        operator: "crossover",
        targetType: "indicator",
        targetIndicator: "SMA",
        targetParams: { period: 200 },
        targetField: "SMA"
      }
    ]
  },
  {
    id: "rsi_macd_fusion",
    name: "Fusion Strategy",
    description: "Safe & Aggressive: RSI < 40 combined with MACD Crossover.",
    side: "buy",
    logicOperator: "AND",
    conditions: [
      {
        id: "fusion-1",
        indicator: "RSI",
        params: { period: 14 },
        operator: "<",
        targetType: "value",
        targetValue: 40
      },
      {
        id: "fusion-2",
        indicator: "MACD",
        params: { fast: 12, slow: 26, signal: 9 },
        field: "MACD",
        operator: "crossover",
        targetType: "indicator",
        targetIndicator: "MACD",
        targetParams: { fast: 12, slow: 26, signal: 9 },
        targetField: "MACDs"
      }
    ]
  },
  {
    id: "bb_reversion",
    name: "BB Mean Reversion",
    description: "Mean reversion: buy when price touches lower Bollinger Band.",
    side: "buy",
    logicOperator: "AND",
    conditions: [
      {
        id: "bb-1",
        indicator: "Price",
        params: {},
        field: "close",
        operator: "<=",
        targetType: "indicator",
        targetIndicator: "BBANDS",
        targetParams: { period: 20, std: 2.0 },
        targetField: "BBL"
      }
    ]
  }
]
